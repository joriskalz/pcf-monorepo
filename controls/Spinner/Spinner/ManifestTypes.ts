import { SpinnerSize } from '@fluentui/react';

export const SpinnerSizes = {
    '3': SpinnerSize.large,
    '2': SpinnerSize.medium,
    '1': SpinnerSize.small,
    '0': SpinnerSize.xSmall,
};

export const SpinnerAlignmentTypes = {
    '0': 'center',
    '1': 'left',
    '2': 'right',
};

export const enum InputEvents {
    SetFocus = 'SetFocus',
}
