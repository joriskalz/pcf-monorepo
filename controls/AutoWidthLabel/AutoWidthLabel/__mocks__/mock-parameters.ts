/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import {
  MockDecimalNumberProperty,
  MockEnumProperty,
  MockStringProperty,
  MockWholeNumberProperty,
} from "./mock-context";

export function getMockParameters(): IInputs {
  return {
    Text: new MockStringProperty(),
    PaddingLeft: new MockWholeNumberProperty(),
    PaddingRight: new MockWholeNumberProperty(),
    PaddingTop: new MockWholeNumberProperty(),
    PaddingBottom: new MockWholeNumberProperty(),
    FontName: new MockStringProperty(),
    FontSize: new MockDecimalNumberProperty(),
    FontSizeUnits: new MockEnumProperty("1"),
    FontColor: new MockStringProperty(),
    FontWeight: new MockStringProperty(),
    DisabledFontColor: new MockStringProperty(),
    DisabledFontWeight: new MockStringProperty(),
    FocusFontColor: new MockStringProperty(),
    FocusFontWeight: new MockStringProperty(),
    HoverFontColor: new MockStringProperty(),
    HoverFontWeight: new MockStringProperty(),
    FillColor: new MockStringProperty(),
    DisabledFillColor: new MockStringProperty(),
    FocusFillColor: new MockStringProperty(),
    HoverFillColor: new MockStringProperty(),
    //DisabledHoverFillColor: new MockStringProperty(),
    BorderColor: new MockStringProperty(),
    BorderThickness: new MockWholeNumberProperty(),
    BorderRadius: new MockWholeNumberProperty(),
    DisabledBorderColor: new MockStringProperty(),
    FocusBorderColor: new MockStringProperty(),
    FocusBorderThickness: new MockWholeNumberProperty(),
    HoverBorderColor: new MockStringProperty(),
    HoverBorderThickness: new MockWholeNumberProperty(),
    //RenderBorderStyle: new MockEnumProperty<"0" | "1">(),
  };
}
