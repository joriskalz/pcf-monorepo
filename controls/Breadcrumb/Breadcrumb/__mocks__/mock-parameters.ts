/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
        MaxDisplayedItems: new MockWholeNumberProperty(),
        OverflowIndex: new MockWholeNumberProperty(),
    };
}
