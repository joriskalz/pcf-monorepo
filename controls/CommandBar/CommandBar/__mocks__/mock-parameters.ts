/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
