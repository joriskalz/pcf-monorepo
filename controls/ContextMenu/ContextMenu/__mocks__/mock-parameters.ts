/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        TextAlignment: new MockEnumProperty(),
        items: new MockDataSet([]),
        Chevron: new MockTwoOptionsProperty(),
        FillColor: new MockStringProperty(),
        BorderColor: new MockStringProperty(),
        BorderRadius: new MockWholeNumberProperty(),
        FontColor: new MockStringProperty(),
        FontSize: new MockWholeNumberProperty(),
        HoverBorderColor: new MockStringProperty(),
        HoverFillColor: new MockStringProperty(),
        HoverFontColor: new MockStringProperty(),
        HoverIconColor: new MockStringProperty(),
        IconColor: new MockStringProperty(),
        IconSize: new MockWholeNumberProperty(),
    };
}
