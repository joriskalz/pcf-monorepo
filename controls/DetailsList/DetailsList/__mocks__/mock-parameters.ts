/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Compact: new MockTwoOptionsProperty(),
        CurrentSortColumn: new MockStringProperty(),
        CurrentSortDirection: new MockEnumProperty('0'),
        PageSize: new MockWholeNumberProperty(),
        RaiseOnRowSelectionChangeEvent: new MockTwoOptionsProperty(),
        SelectRowsOnFocus: new MockTwoOptionsProperty(),
        SelectionType: new MockEnumProperty('0'),
        AlternateRowColor: new MockStringProperty(),
        HeaderVisible: new MockTwoOptionsProperty(false),
        SelectionAlwaysVisible: new MockTwoOptionsProperty(false),
        records: new MockDataSet([]),
        columns: new MockDataSet([]),
        LargeDatasetPaging: new MockTwoOptionsProperty(),
    };
}
