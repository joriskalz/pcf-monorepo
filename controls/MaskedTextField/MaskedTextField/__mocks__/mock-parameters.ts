/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty } from './mock-context';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Value: new MockStringProperty(),
        Prefix: new MockStringProperty(),
        Suffix: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        ErrorMessage: new MockStringProperty(),
        Mask: new MockStringProperty(),
        MaskFormat: new MockStringProperty(),
        DelayOutput: new MockTwoOptionsProperty(),
    };
}
