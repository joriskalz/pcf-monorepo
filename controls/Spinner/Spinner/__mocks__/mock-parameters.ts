/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty } from './mock-context';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        Label: new MockStringProperty(),
        SpinnerSize: new MockEnumProperty(),
        LabelPosition: new MockEnumProperty(),
        SpinnerAlignment: new MockEnumProperty(),
        BackgroundColor: new MockStringProperty(),
    };
}
