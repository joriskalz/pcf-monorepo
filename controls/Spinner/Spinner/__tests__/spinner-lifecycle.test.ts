/* eslint-disable sonarjs/no-identical-functions */
import { Spinner } from '..';
import { IInputs } from '../generated/ManifestTypes';
import { MockContext, MockState } from '../__mocks__/mock-context';
import { getMockParameters } from '../__mocks__/mock-parameters';
import * as renderer from 'react-test-renderer';

// Since requestAnimationFrame does not exist in the test DOM, mock it
window.requestAnimationFrame = jest.fn().mockImplementation((callback) => {
    callback();
});

describe('Spinner', () => {
    beforeEach(() => {
        jest.clearAllMocks();
        jest.useFakeTimers();
    });
    afterEach(() => {
        for (let i = 0; i < document.body.children.length; i++) {
            if (document.body.children[i].tagName === 'DIV') {
                document.body.removeChild(document.body.children[i]);
                i--;
            }
        }
        // reset any jest timers
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        if ((setTimeout as any).mock) {
            jest.runOnlyPendingTimers();
            jest.useRealTimers();
        }
    });

    it('theme', async () => {
        const { component, context } = createComponent();
        context.parameters.Theme.raw = JSON.stringify({
            palette: {
                themePrimary: '#test-primary',
            },
        });

        component.init(context);
        const spinnerComponent = renderer.create(component.updateView(context));
        expect(spinnerComponent.toJSON()).toMatchSnapshot();
    });

    it('test SpinnerSize', async () => {
        const { component, context } = createComponent();
        component.init(context);
        context.parameters.SpinnerSize.raw = '0';
        let spinnerComponent = renderer.create(component.updateView(context));
        expect(spinnerComponent.toJSON()).toMatchSnapshot();
        context.parameters.SpinnerSize.raw = '1';
        spinnerComponent = renderer.create(component.updateView(context));
        expect(spinnerComponent.toJSON()).toMatchSnapshot();
        context.parameters.SpinnerSize.raw = '2';
        spinnerComponent = renderer.create(component.updateView(context));
        expect(spinnerComponent.toJSON()).toMatchSnapshot();
        context.parameters.SpinnerSize.raw = '3';
        spinnerComponent = renderer.create(component.updateView(context));
        expect(spinnerComponent.toJSON()).toMatchSnapshot();
    });
});

function createComponent() {
    const component = new Spinner();
    const notifyOutputChanged = jest.fn();
    const context = new MockContext<IInputs>(getMockParameters());
    context.parameters.Label.raw = 'Loading...';
    context.parameters.AccessibilityLabel.raw = 'Spinner';
    context.parameters.SpinnerSize.raw = '2';
    context.parameters.LabelPosition.raw = 'Top';
    const state = new MockState();
    const container = document.createElement('div');
    document.body.appendChild(container);
    return { component, context, notifyOutputChanged, container, state };
}
