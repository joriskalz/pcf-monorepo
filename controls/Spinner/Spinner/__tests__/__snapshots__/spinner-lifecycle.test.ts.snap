// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner test SpinnerSize 1`] = `
<div
  className=

      {
        display: flex;
        height: -1px;
        width: -1px;
      }
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background: #ffffff;
        color: #323130;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
>
  <div
    aria-label="Spinner"
    className=
        ms-Spinner
        {
          align-items: center;
          display: flex;
          flex-direction: column-reverse;
          justify-content: center;
        }
  >
    <div
      className=
          ms-Spinner-circle
          ms-Spinner--xSmall
          {
            animation-duration: 1.3s;
            animation-iteration-count: infinite;
            animation-name: keyframes 0%{transform:rotate(0deg);}100%{transform:rotate(360deg);};
            animation-timing-function: cubic-bezier(.53,.21,.29,.67);
            border-radius: 50%;
            border-top-color: #0078d4;
            border: 1.5px solid #c7e0f4;
            box-sizing: border-box;
            height: 12px;
            width: 12px;
          }
          @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
            -ms-high-contrast-adjust: none;
            border-top-color: Highlight;
            forced-color-adjust: none;
          }
    />
    <div
      className=
          ms-Spinner-label
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: #0078d4;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 12px;
            font-weight: 400;
            margin-bottom: 8px;
            margin-left: 0;
            margin-right: 0;
            margin-top: 0;
            text-align: center;
          }
    >
      Loading...
    </div>
  </div>
</div>
`;

exports[`Spinner test SpinnerSize 2`] = `
<div
  className=

      {
        display: flex;
        height: -1px;
        width: -1px;
      }
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background: #ffffff;
        color: #323130;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
>
  <div
    aria-label="Spinner"
    className=
        ms-Spinner
        {
          align-items: center;
          display: flex;
          flex-direction: column-reverse;
          justify-content: center;
        }
  >
    <div
      className=
          ms-Spinner-circle
          ms-Spinner--small
          {
            animation-duration: 1.3s;
            animation-iteration-count: infinite;
            animation-name: keyframes 0%{transform:rotate(0deg);}100%{transform:rotate(360deg);};
            animation-timing-function: cubic-bezier(.53,.21,.29,.67);
            border-radius: 50%;
            border-top-color: #0078d4;
            border: 1.5px solid #c7e0f4;
            box-sizing: border-box;
            height: 16px;
            width: 16px;
          }
          @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
            -ms-high-contrast-adjust: none;
            border-top-color: Highlight;
            forced-color-adjust: none;
          }
    />
    <div
      className=
          ms-Spinner-label
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: #0078d4;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 12px;
            font-weight: 400;
            margin-bottom: 8px;
            margin-left: 0;
            margin-right: 0;
            margin-top: 0;
            text-align: center;
          }
    >
      Loading...
    </div>
  </div>
</div>
`;

exports[`Spinner test SpinnerSize 3`] = `
<div
  className=

      {
        display: flex;
        height: -1px;
        width: -1px;
      }
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background: #ffffff;
        color: #323130;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
>
  <div
    aria-label="Spinner"
    className=
        ms-Spinner
        {
          align-items: center;
          display: flex;
          flex-direction: column-reverse;
          justify-content: center;
        }
  >
    <div
      className=
          ms-Spinner-circle
          ms-Spinner--medium
          {
            animation-duration: 1.3s;
            animation-iteration-count: infinite;
            animation-name: keyframes 0%{transform:rotate(0deg);}100%{transform:rotate(360deg);};
            animation-timing-function: cubic-bezier(.53,.21,.29,.67);
            border-radius: 50%;
            border-top-color: #0078d4;
            border: 1.5px solid #c7e0f4;
            box-sizing: border-box;
            height: 20px;
            width: 20px;
          }
          @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
            -ms-high-contrast-adjust: none;
            border-top-color: Highlight;
            forced-color-adjust: none;
          }
    />
    <div
      className=
          ms-Spinner-label
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: #0078d4;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 12px;
            font-weight: 400;
            margin-bottom: 8px;
            margin-left: 0;
            margin-right: 0;
            margin-top: 0;
            text-align: center;
          }
    >
      Loading...
    </div>
  </div>
</div>
`;

exports[`Spinner test SpinnerSize 4`] = `
<div
  className=

      {
        display: flex;
        height: -1px;
        width: -1px;
      }
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background: #ffffff;
        color: #323130;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
>
  <div
    aria-label="Spinner"
    className=
        ms-Spinner
        {
          align-items: center;
          display: flex;
          flex-direction: column-reverse;
          justify-content: center;
        }
  >
    <div
      className=
          ms-Spinner-circle
          ms-Spinner--large
          {
            animation-duration: 1.3s;
            animation-iteration-count: infinite;
            animation-name: keyframes 0%{transform:rotate(0deg);}100%{transform:rotate(360deg);};
            animation-timing-function: cubic-bezier(.53,.21,.29,.67);
            border-radius: 50%;
            border-top-color: #0078d4;
            border: 1.5px solid #c7e0f4;
            box-sizing: border-box;
            height: 28px;
            width: 28px;
          }
          @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
            -ms-high-contrast-adjust: none;
            border-top-color: Highlight;
            forced-color-adjust: none;
          }
    />
    <div
      className=
          ms-Spinner-label
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: #0078d4;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 12px;
            font-weight: 400;
            margin-bottom: 8px;
            margin-left: 0;
            margin-right: 0;
            margin-top: 0;
            text-align: center;
          }
    >
      Loading...
    </div>
  </div>
</div>
`;

exports[`Spinner theme 1`] = `
<div
  className=

      {
        display: flex;
        height: -1px;
        width: -1px;
      }
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background: #ffffff;
        color: #323130;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
>
  <div
    aria-label="Spinner"
    className=
        ms-Spinner
        {
          align-items: center;
          display: flex;
          flex-direction: column-reverse;
          justify-content: center;
        }
  >
    <div
      className=
          ms-Spinner-circle
          ms-Spinner--medium
          {
            animation-duration: 1.3s;
            animation-iteration-count: infinite;
            animation-name: keyframes 0%{transform:rotate(0deg);}100%{transform:rotate(360deg);};
            animation-timing-function: cubic-bezier(.53,.21,.29,.67);
            border-radius: 50%;
            border-top-color: #test-primary;
            border: 1.5px solid #c7e0f4;
            box-sizing: border-box;
            height: 20px;
            width: 20px;
          }
          @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
            -ms-high-contrast-adjust: none;
            border-top-color: Highlight;
            forced-color-adjust: none;
          }
    />
    <div
      className=
          ms-Spinner-label
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: #test-primary;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 12px;
            font-weight: 400;
            margin-bottom: 8px;
            margin-left: 0;
            margin-right: 0;
            margin-top: 0;
            text-align: center;
          }
    >
      Loading...
    </div>
  </div>
</div>
`;
